imod/typing/grid.py

```diff
@@ -306,3 +306,20 @@ def get_spatial_dimension_names( grid: xu.UgridDataArray)-> list[str]:
 @typedispatch
 def get_spatial_dimension_names( grid: object)-> list[str]:
     return []
+
+@typedispatch
+def get_grid_geometry_hash( grid: xr.DataArray)-> int:
+    result = hash(pickle.dumps(grid["x"].values))
```

Comment:
It's worth researching/checking how fast the different options are for hashing numpy arrays. See for example: https://stackoverflow.com/questions/16589791/most-efficient-property-to-hash-for-numpy-array

I've thought about xxhash before, but I'm not sure it's worth it to add another dependency just for this use.

I don't think it's necessary to use cryptographic hashes from hashlib, but it's probably worth it to recheck what kind of algorithm `hash` is using behind the scenes.

Reply:
We could try and see if xxhash provides any significant performance gains for us. 

Installation wise:
[The python bindings are built to Windows, Linux, and Mac on conda-forge](https://anaconda.org/conda-forge/python-xxhash/). The [feedstock has two maintainers](https://github.com/conda-forge/python-xxhash-feedstock), but luckily they are very active.  So I don't foresee it making our installation process much harder than it already is.

Reply:
let's first get it working like this, and measure if hashing is a time sink for large models. If it is, then we can try xxhash.